pkg/external_tools/mcp/utils.go

```diff
@@ -0,0 +1,151 @@
+	k8stypes "github.com/ovn-kubernetes/ovn-kubernetes-mcp/pkg/kubernetes/types"
+)
+
+var interfaceNamePattern = regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9._-]*$`)
+
+// runDebugNode runs a command on a node
+func (s *MCPServer) runDebugNode(ctx context.Context, req *mcp.CallToolRequest, target types.DebugNodeParams, command []string) (*mcp.CallToolResult, types.CommandResult, error) {
+	if target.NodeName == "" {
+		return nil, types.CommandResult{}, fmt.Errorf("node_name is required when target_type is 'node'")
+	}
+	if target.NodeImage == "" {
+		return nil, types.CommandResult{}, fmt.Errorf("node_image is required when target_type is 'node'")
+	}
+	_, output, err := s.k8sMcpServer.DebugNode(ctx, req, k8stypes.DebugNodeParams{
+		Name:      target.NodeName,
+		Image:     target.NodeImage,
+		Command:   command,
+		HostPath:  target.HostPath,
+		MountPath: target.MountPath,
+	})
+	if err != nil {
+		return nil, types.CommandResult{}, err
+	}
+	return nil, types.CommandResult{Output: output.Stdout}, nil
+}
+
+// runExecPod runs a command on a pod
+func (s *MCPServer) runExecPod(ctx context.Context, req *mcp.CallToolRequest, target types.ExecPodParams, command []string) (*mcp.CallToolResult, types.CommandResult, error) {
+	if target.PodName == "" {
+		return nil, types.CommandResult{}, fmt.Errorf("pod_name is required when target_type is 'pod'")
+	}
+	if target.PodNamespace == "" {
+		target.PodNamespace = "default"
+	}
+	_, output, err := s.k8sMcpServer.ExecPod(ctx, req, k8stypes.ExecPodParams{
+		NamespacedNameParams: k8stypes.NamespacedNameParams{
+			Name:      target.PodName,
+			Namespace: target.PodNamespace,
+		},
+		Container: target.ContainerName,
+		Command:   command,
+	})
+	if err != nil {
+		return nil, types.CommandResult{}, err
+	}
+	return nil, types.CommandResult{Output: output.Stdout}, nil
+}
+
+// commandBuilder helps build commands with a fluent interface
+type commandBuilder struct {
+	args []string
+}
+
+// newCommand creates a new command builder with the base command
+func newCommand(baseCmd ...string) *commandBuilder {
+	return &commandBuilder{args: baseCmd}
+}
+
+// add adds arguments to the command
+func (cb *commandBuilder) add(args ...string) *commandBuilder {
+	cb.args = append(cb.args, args...)
+	return cb
+}
+
+// addIf adds arguments to the command only if the condition is true
+func (cb *commandBuilder) addIf(condition bool, args ...string) *commandBuilder {
+	if condition {
+		cb.args = append(cb.args, args...)
+	}
+	return cb
+}
+
+// addIfNotEmpty adds arguments to the command only if the value is not empty
+func (cb *commandBuilder) addIfNotEmpty(value string, args ...string) *commandBuilder {
+	if value != "" {
+		cb.args = append(cb.args, args...)
+	}
+	return cb
+}
+
+// build returns the final command slice
+func (cb *commandBuilder) build() []string {
+	return cb.args
+}
+
+// ValidateInterface validates a network interface name for security and correctness.
+func ValidateInterface(iface string) error {
```

Comment:
```suggestion
// validateInterface validates a network interface name for security and correctness.
func validateInterface(iface string) error {
```
Since this function is only going to be used in this package, let's not export it. Same goes for the other fuctions in this file.